pkg/analysis/numericbounds/testdata/src/a/a.go

```diff
@@ -0,0 +1,238 @@
+}
+
+// IgnoredStringField should not be checked
+type IgnoredStringField struct {
+	Name string
+}
+
+// IgnoredBoolField should not be checked
+type IgnoredBoolField struct {
+	Enabled bool
+}
+
+// ValidFloat64WithBounds has proper bounds validation
+type ValidFloat64WithBounds struct {
+	// +kubebuilder:validation:Minimum=-1000.5
+	// +kubebuilder:validation:Maximum=1000.5
+	Value float64
+}
+
+// ValidFloat32WithBounds has proper bounds validation
+type ValidFloat32WithBounds struct {
+	// +kubebuilder:validation:Minimum=0.0
+	// +kubebuilder:validation:Maximum=100.0
+	Ratio float32
+}
+
+// InvalidFloat64NoBounds should have bounds markers
+type InvalidFloat64NoBounds struct {
+	Value float64 // want "InvalidFloat64NoBounds.Value is missing minimum bound validation marker" "InvalidFloat64NoBounds.Value is missing maximum bound validation marker"
+}
+
+// InvalidFloat32NoBounds should have bounds markers
+type InvalidFloat32NoBounds struct {
+	Ratio float32 // want "InvalidFloat32NoBounds.Ratio is missing minimum bound validation marker" "InvalidFloat32NoBounds.Ratio is missing maximum bound validation marker"
+}
+
+// MixedFields has both valid and invalid fields
+type MixedFields struct {
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=100
+	ValidCount int32
+
+	InvalidCount int32 // want "MixedFields.InvalidCount is missing minimum bound validation marker" "MixedFields.InvalidCount is missing maximum bound validation marker"
+
+	Name string
+
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=1000
+	ValidValue int64
+
+	InvalidValue int64 // want "MixedFields.InvalidValue is missing minimum bound validation marker" "MixedFields.InvalidValue is missing maximum bound validation marker"
+}
+
+// PointerFields with pointers should also be checked
+type PointerFields struct {
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=100
+	ValidPointerWithBounds *int32
+
+	InvalidPointer *int32 // want "PointerFields.InvalidPointer is missing minimum bound validation marker" "PointerFields.InvalidPointer is missing maximum bound validation marker"
+
+	InvalidPointer64 *int64 // want "PointerFields.InvalidPointer64 is missing minimum bound validation marker" "PointerFields.InvalidPointer64 is missing maximum bound validation marker"
+}
+
+// SliceFields with slices should check the element type
+type SliceFields struct {
+	ValidSlice []string
+
+	InvalidSlice []int32 // want "SliceFields.InvalidSlice is missing minimum bound validation marker" "SliceFields.InvalidSlice is missing maximum bound validation marker"
+
+	InvalidSlice64 []int64 // want "SliceFields.InvalidSlice64 is missing minimum bound validation marker" "SliceFields.InvalidSlice64 is missing maximum bound validation marker"
+}
+
+// TypeAliasFields with type aliases should be checked
+type Int32Alias int32
+type Int64Alias int64
+type Float32Alias float32
+type Float64Alias float64
```

Comment:
Should we also be checking types, not just fields? This would align with other type checker implementations